java/main/src/main/java/com/saucelabs/saucebindings/performance/Performance.java

```diff
@@ -0,0 +1,131 @@
+     */
+    public PerformanceResults getResults(List<String> metrics) {
+        HashMap<String, Object> perf = new HashMap<>();
+        perf.put("name", session.getSauceOptions().sauce().getName());
+        perf.put("metrics", metrics);
+        Map<String, Object> performance = (Map<String, Object>) session.getDriver().executeScript("sauce:performance", perf);
+        return new PerformanceResults(performance);
+    }
+
+    /**
+     * @return JenkinsResults is a wrapper of the HashMap results to make it easier to work with them
+     * @see <a href="https://docs.saucelabs.com/performance/motion">Measuring On-Page Motion Effects</a>
+     */
+    public JankinessResults getJankinessResults() {
+        Map<String, Object> jankiness = (Map<String, Object>) session.getDriver().executeScript("sauce:jankinessCheck");
+        return new JankinessResults(jankiness);
+    }
+
+    /**
+     * This obtains information on all of the collected metrics to allow making assertions on one or more values
+     *
+     * @return PerformanceMetrics is a wrapper of the HashMap results to make it easier to work with them
+     * @see <a href="https://docs.saucelabs.com/performance/transitions#logging-performance-results">Logging Performance Results</a>
+     */
+    public PerformanceMetrics getMetrics() {
+        HashMap<String, Object> metricsLog = new HashMap<>();
+        metricsLog.put("type", "sauce:performance");
+
+        Map<String, Object> metrics = (Map<String, Object>) session.getDriver().executeScript("sauce:log", metricsLog);
+        return new PerformanceMetrics(metrics);
+    }
+
+    /**
+     * Turn on recording of metrics during session
+     * @see <a href="https://docs.saucelabs.com/performance/transitions#target-specific-urls-in-a-script">Target Specific URLs in a Script</a>
+     */
+    public void enable() {
+        session.getDriver().executeScript("sauce:performanceEnable");
+    }
+
+    /**
+     * Turn off recording of metrics during session
+     * @see <a href="https://docs.saucelabs.com/performance/transitions#target-specific-urls-in-a-script">Target Specific URLs in a Script</a>
+     */
+    public void disable() {
+        session.getDriver().executeScript("sauce:performanceDisable");
+    }
+
+    /**
+     * The map used as input will typically be converted from serialized information stored in a yaml or JSON file
+     * @param budget Map of Maps with the URL as the keys of outer maps, the metric as the keys of the nested maps and the maximum value
+     *               of the metric as its value
+     * @return A Map of failures if any, including the failing metric and its value
+     * @see <a href="https://docs.saucelabs.com/performance/transitions#defining-a-performance-budget">Defining a Performance Budget</a>
+     */
+    public Map<String, Object> getBudgetFailures(Map<String, Map<String, Object>> budget) {
```

Comment:
Not sure I understand what this method does when first reading it. The term `budget` really throws me off as I've never encountered it with front-end perf.
Is it meant to be public?

Reply:
Yeah, look at the link to our performance docs that I put in javadocs which discusses it. Christian implemented it for wdio, and showed JS code. Just trying to keep up with wdio. :)